ocpp-common/src/main/java/eu/chargetime/ocpp/Session.java

```diff
@@ -231,6 +238,36 @@ public void onConnected() {
     }
   }
 
+  private void addPendingPromise(String id, String action, CompletableFuture<Confirmation> promise) {
+    synchronized (pendingPromises) {
+      pendingPromises.put(id, new AbstractMap.SimpleImmutableEntry<>(action, promise));
+    }
+  }
+
+  @Override
+  public boolean completePendingPromise(String id, Confirmation confirmation) throws UnsupportedFeatureException, OccurenceConstraintException {
```

Comment:
It may return false if it did not find pending promise:

https://github.com/ChargeTimeEU/Java-OCA-OCPP/blob/b6c0b590e2f180b2518f053a5ce846dbd6d9c748/ocpp-common/src/main/java/eu/chargetime/ocpp/Session.java#L251-L256